src/AbstractUser.java

```diff
@@ -0,0 +1,23 @@
+import java.util.ArrayList;
+
+public class AbstractUser implements IUser {

```

Comment:
IUser is a bad idea here to use as something that can have inventory. Think more global.
In you game many different entities can have the functionality to have an inventory. For example Player can have it, but also in game character my have it. They might be different classes, but they will still implement some similar interfaces. maybe even more. While your user still may implement IUser interface for things like "jump", "die" or other that are specific for all types of users. In this case you are interested in interface that is Meant to provide functionality to work with inventory. It's good to separate this. And re-use later.

Create IInventoryProvider interface something that has methods like getMaterials, subsctract, add and so on.
And then your IUser in this case can be even empty for now, and AbstractUser implements IUser, IInventoryProvider

(two interfaces)

If you put everything in one IUser interface yo uwill never be able to implement it by any other class and one off interfaces make no sense (hena miangamic prosto useri mej kgreir eli et sagh metodner@)